### PR TITLE
Reference manual: Show modularized sagelib packages separately

### PR DESCRIPTION
**Pull request by mkoeppe**
### :books: Description

Packages such as **sagemath-categories** are currently mixed with optional SPKGs in the reference guide.
We separate them out as a separate category.

We also move the section on `Feature`s right next to the package list.
#35749 and follow-ups will interlink these sections.

Preview:
- https://deploy-preview-35734--sagemath-tobias.netlify.app/reference/spkg/index.html#packages-of-the-modularized-sage-library
- https://deploy-preview-35734--sagemath-tobias.netlify.app/reference/spkg/index.html#runtime-detectable-features-and-conditional-doctests

- Part of #29705 

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/features/kenzo.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 class Kenzo(Feature):
     r"""
-    A :class:`~sage.features.Feature` describing the presence of ``Kenzo``.
+    A :class:`~sage.features.Feature` describing the presence of :class:`Kenzo <spkg_kenzo>`.
```
Comment (kwankyu) [inline src/sage/features/kenzo.py:23]: `:ref:`?

File: src/sage/features/latte.py [hunk 2]
```diff
@@ -58,8 +58,7 @@ def __init__(self):
 
 class Latte(JoinFeature):
     r"""
-    A :class:`~sage.features.Feature` describing the presence of the ``LattE``
-    binaries which comes as a part of ``latte_int``.
+    A :class:`~sage.features.Feature` describing the presence of excecutables from :ref:`LattE integrale <spkg_latte_int>`.
```
Comment (kwankyu) [inline src/sage/features/latte.py:63]: PEP8.

File: src/sage/features/graphviz.py [hunk 3]
```diff
@@ -89,8 +89,8 @@ def __init__(self):
 class Graphviz(JoinFeature):
     r"""
     A :class:`~sage.features.Feature` describing the presence of
-    the ``dot``, ``neato``, and ``twopi`` executables from the
-    ``graphviz`` package.
+    the :class:`dot`, :class:`neato`, and :class:`twopi` executables from the
+    :ref:`graphviz <spkg_graphviz>` package.
```
Comment (kwankyu) [inline src/sage/features/graphviz.py:93]: Why `:class:`?

Comment (mkoeppe) [reply src/sage/features/graphviz.py:93]: these are Features

File: src/doc/bootstrap [hunk 4]
```diff
@@ -100,7 +100,16 @@ Sage depends.  It installs them automatically if it does not find
...
 EOF
-for PKG_BASE in $(sage-package list --has-file SPKG.rst :standard: | sort); do
+for PKG_BASE in $(sage-package list --has-file SPKG.rst :standard: | grep -v '^sagemath_' | sort); do
+    echo "* :ref:\`spkg_$PKG_BASE\`"
+done >> "$OUTPUT_INDEX"
+cat >> "$OUTPUT_INDEX" <<EOF
+
+Packages of the Modularized Sage Library
+----------------------------------------
+
```
Comment (kwankyu) [inline src/doc/bootstrap]: How about "Distribution Packages of the Sage Library", now that "distribution package" is a standard term?

Comment (mkoeppe) [reply src/doc/bootstrap]: Good idea, done

File: src/doc/bootstrap [hunk 5]
```diff
@@ -125,11 +134,48 @@ Some packages that provide additional functionality are marked as
...
 
 EOF
-for PKG_BASE in $(sage-package list --has-file SPKG.rst :experimental: | sort); do
+for PKG_BASE in $(sage-package list --has-file SPKG.rst :experimental: | grep -v '^sagemath_' | sort); do
     echo "* :ref:\`spkg_$PKG_BASE\`"
 done >> "$OUTPUT_INDEX"
 cat >> "$OUTPUT_INDEX" <<EOF
 
+Runtime-Detectable Features and Conditional Doctests
+----------------------------------------------------
```
Comment (kwankyu) [inline src/doc/bootstrap]: How about just "Features and Conditional Doctests" or "Features for Conditional Doctests" or even just "Features"? 

We do not need to explain what a feature is in the title. Or do you need? If not, let's make the title short.

File: src/doc/bootstrap [hunk 6]
```diff
@@ -125,11 +134,48 @@ Some packages that provide additional functionality are marked as
...
+   sage/features/mip_backends
+   sage/features/normaliz
+   sage/features/pandoc
+   sage/features/pdf2svg
+   sage/features/polymake
+   sage/features/rubiks
+   sage/features/tdlib
+   sage/features/join_feature
+   sage/features/all
+
```
Comment (kwankyu) [inline src/doc/bootstrap:181]: How about splitting feature facilities from features themselves? So `sage/features`, `sage/features/join_feature`, `sage/features/all` may be placed fore-front and then features afterwards.

Comment (mkoeppe) [reply src/doc/bootstrap:181]: In a separate toctree then?

File: src/doc/bootstrap (hunk 5 above)
Comment (mkoeppe) [reply src/doc/bootstrap]: I've shortened it

File: src/doc/bootstrap [hunk 7]
```diff
@@ -89,8 +89,8 @@ fi
 OUTPUT_INDEX="$OUTPUT_DIR"/index.rst
 cat > "$OUTPUT_INDEX" <<EOF
 
-External Packages
-=================
+Features, Distributions, and External Packages
+==============================================
```
Comment (kwankyu) [inline src/doc/bootstrap:94]: How about just "Packages and Features"?

And we list them in the order

- standard packages
- optional packages
- distribution packages
- features
- experimental packages (perhaps after features)
- all packages (I don't know what to do with this item. Maybe removed later)

File: src/doc/bootstrap (hunk 6 above)
Comment (kwankyu) [reply src/doc/bootstrap:181]: Maybe. Perhaps we may later move them three out of this section if we could find a better place for them. But I have no idea right now.

File: src/sage/features/graphviz.py (hunk 3 above)
Comment (kwankyu) [reply src/sage/features/graphviz.py:93]: OK.

File: src/doc/bootstrap (hunk 6 above)
Comment (kwankyu) [reply src/doc/bootstrap:181]: A good place for the three may be found in the section

Infrastructure > Programming Facilities > Test Framework

or a new section

Infrastructure > Programming Facilities > Features Framework (or Features Facility or just Features)

**Review by kwankyu — APPROVED** [review]
Thanks. LGTM.

File: src/doc/bootstrap (hunk 6 above)
Comment (mkoeppe) [reply src/doc/bootstrap:181]: Right, that's where they are coming from...

**Comment by mkoeppe** [comment]
Thank you!

**Pull request merged**